python/ray/_private/runtime_env/packaging.py

```diff
@@ -727,13 +732,16 @@ def open_file(uri, mode, *, transport_params=None):
                     with open_file(pkg_file, "wb") as fin:
                         fin.write(package_zip.read())
 
-                unzip_package(
-                    package_path=pkg_file,
-                    target_dir=local_dir,
-                    remove_top_level_directory=True,
-                    unlink_zip=True,
-                    logger=logger,
-                )
+                if Path(pkg_name).suffix == ".zip":
+                    unzip_package(
+                        package_path=pkg_file,
+                        target_dir=local_dir,
+                        remove_top_level_directory=True,
+                        unlink_zip=True,
+                        logger=logger,
+                    )
+                else:
```

Comment:
What cases are handled by this "else", is it just ".whl" or are there other possibilities? Should we raise an error if we have an unexpected suffix?

Reply:
Yes, just whl, gonna change it to raise Error